From 2d5a03cf61a19029343afc90bdabf5dd661f423e Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:16:15 -0400
Subject: [net] netfilter: xt_log: add missing string format in nf_log_packet()

Message-id: <20141021161615.22888.30643.stgit@localhost.localdomain>
Patchwork-id: 97911
O-Subject: [RHEL7 net PATCH 014/124] netfilter: xt_LOG: add missing string format in nf_log_packet()
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: ca1aa54f272d47bec77baa292f803df7a81f966b

 commit ca1aa54f272d47bec77baa292f803df7a81f966b
 Author: Pablo Neira Ayuso <pablo@netfilter.org>
 Date:   Sat Jun 28 18:42:41 2014 +0200

    netfilter: xt_LOG: add missing string format in nf_log_packet()

       net/netfilter/xt_LOG.c: In function 'log_tg':
    >> net/netfilter/xt_LOG.c:43: error: format not a string literal and no format arguments

    Fixes: fab4085 ("netfilter: log: nf_log_packet() as real unified interface")
    Reported-by: Fengguang Wu <fengguang.wu@intel.com>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netfilter/xt_LOG.c b/net/netfilter/xt_LOG.c
index 00eb491..c13b794 100644
--- a/net/netfilter/xt_LOG.c
+++ b/net/netfilter/xt_LOG.c
@@ -40,7 +40,7 @@ log_tg(struct sk_buff *skb, const struct xt_action_param *par)
  li.u.log.logflags = loginfo->logflags;
 
  nf_log_packet(net, par->family, par->hooknum, skb, par->in, par->out,
-        &li, loginfo->prefix);
+        &li, "%s", loginfo->prefix);
  return XT_CONTINUE;
 }
 
-- 
1.7.1